internal/core/application/trade_service.go

```diff
@@ -105,6 +108,40 @@ func newTradeService(
 		feeAccountBalanceThreshold: feeAccountBalanceThreshold,
 		lock:                       &sync.Mutex{},
 	}
+
+	if skipCheckTrades {
+		return svc
+	}
+	// In case of restart, restore watching trades/outpoints: start observing the
+	// inputs of each trade even if it's already expired.
+	// The idea is to wait at least for a minute to give the time to the bc
+	// listener to make sure the trade has been settled.
+	// Otherwise, the trade will expire and the utxos will be unlocked.
+	trades, _ := svc.repoManager.TradeRepository().GetAllTrades(context.Background())
+	for i := range trades {
+		trade := trades[i]
+		if !trade.Status.Failed && trade.IsAccepted() {
+			ptx, _ := pset.NewPsetFromBase64(trade.PsetBase64)
```

Comment:
should we put all from L124 - L140 in another goroutine? otherwise we need to wait atleast one min for each accepted trade.